lib/helpers/AxiosURLSearchParams.js

```diff
@@ -39,20 +39,20 @@ function AxiosURLSearchParams(params, options) {
   params && toFormData(params, this, options);
 }
 
-var prototype = AxiosURLSearchParams.prototype;
-
-prototype.append = function append(name, value) {
-  this._pairs.push([name, value]);
-};
-
-prototype.toString = function toString(encoder) {
-  var _encode = encoder ? function(value) {
-    return encoder.call(this, value, encode);
-  } : encode;
-
-  return this._pairs.map(function each(pair) {
-    return _encode(pair[0]) + '=' + _encode(pair[1]);
-  }, '').join('&');
+AxiosURLSearchParams.prototype = {
```

Comment:
Rewriting the function prototype handle is an old ES3 antipattern that breaks the proper inheritance- no valid proto chaining and no valid constructor in the new prototype- instance.constructor refers to the Object constructor, not AxiosURLSearchParams.

Reply:
Interesting. Do you have a source where I can read more about this?